src/Compilers/CSharp/Test/CommandLine/SarifV2ErrorLoggerTests.cs

```diff
@@ -651,5 +651,628 @@ class C
             CleanupAllGeneratedFiles(sourceFile);
             CleanupAllGeneratedFiles(errorLogFile);
         }
+
+        public enum SarifTestVersion { V1, V2 }
+
+        private string GetErrorLogQualifier(SarifTestVersion version)
+        {
+            return version == SarifTestVersion.V1 ? "" : ";version=2";
+        }
+
+        [ConditionalTheory(typeof(WindowsOnly), Reason = "https://github.com/dotnet/roslyn/issues/30289")]
+        [CombinatorialData]
+        public void PathMap(SarifTestVersion version)
+        {
+            var source = @"
+public class C
+{
+    private int x;
+}";
+            var tempDir = Temp.CreateDirectory();
+            var errorLogFile = Path.Combine(tempDir.Path, "ErrorLog.txt");
+            var sourceFile = tempDir.CreateFile("file.cs").WriteAllText(source).Path;
+
+            string[] arguments = new[] { "/nologo", sourceFile, "/preferreduilang:en", $"/errorlog:{errorLogFile}{GetErrorLogQualifier(version)}", $"/pathmap:{tempDir}=D:/mypath" };
+
+            var cmd = CreateCSharpCompiler(null, WorkingDirectory, arguments);
+            var outWriter = new StringWriter(CultureInfo.InvariantCulture);
+
+            var exitCode = cmd.Run(outWriter);
+            var actualConsoleOutput = outWriter.ToString().Trim();
+
+            Assert.Contains("CS0169", actualConsoleOutput);
+            Assert.Contains("CS5001", actualConsoleOutput);
+            Assert.NotEqual(0, exitCode);
+
+            var actualOutput = File.ReadAllText(errorLogFile).Trim();
+            if (version == SarifTestVersion.V1)
+            {
+                AssertEx.EqualOrDiff("""
+                    {
+                      "$schema": "http://json.schemastore.org/sarif-1.0.0",
+                      "version": "1.0.0",
+                      "runs": [
+                        {
+                          "tool": {
+                            "name": "Microsoft (R) Visual C# Compiler",
+                            "version": "42.42.42.42",
+                            "fileVersion": "4.9.0-dev (<developer build>)",
+                            "semanticVersion": "42.42.42",
```

Comment:
Can't hard code these values cause they will change. Even the version will change between local and CI (-dev vs. -ci). Need to either filter these out or fill them with interpolation.